crates/holochain_sqlite/src/db/p2p_agent_store.rs

```diff
@@ -54,7 +54,7 @@ pub trait AsP2pStateTxExt {
     fn p2p_query_near_basis(&self, basis: u32, limit: u32) -> DatabaseResult<Vec<AgentInfoSigned>>;
 }
 
-impl AsP2pAgentStoreConExt for crate::db::PConn {
+impl AsP2pAgentStoreConExt for crate::db::PConnGuard {
```

Comment:
In regards to my comment about different types...perhaps here is where you'd split these methods into different traits depending on what kind of connection you want to use with them.

Reply:
Doesn't this already force them to use the `PConnGuard` which can only be created via a permit?